src/sys/socket/mod.rs

```diff
@@ -1403,6 +1541,16 @@ impl<'a> ControlMessage<'a> {
             ControlMessage::TxTime(_) => {
                 libc::SCM_TXTIME
             },
+            #[cfg(any(linux_android, target_os = "freebsd"))]
+            #[cfg(feature = "net")]
+            ControlMessage::Ipv4Tos(_) => {
+                libc::IP_TOS
+            },
```

Comment:
I think you mentioned to return `libc::IP_TOS` on Linux and `libc::IP_RECVTOS` on freebsd, but the tests return `ENOPROTO` on Freebsd if we do this, while they work properly if we return `libc::IP_TOS` for both os flavours.

Reply:
Just rechecked the FreeBSD manual, looks that the IP_RECVXXX is only used with recvmsg(). 

Yeah, FreeBSD should use the same cmsg_type when it comes to ControlMessage, let's update this for both Ipv4Ttl and Ipv4Tos.


Sorry for being misleading:<